addons/stock_landed_costs/models/stock_landed_cost.py

```diff
@@ -289,6 +279,38 @@ def action_view_stock_valuation_layers(self):
         action = self.env.ref('stock_account.stock_valuation_layer_action').read()[0]
         return dict(action, domain=domain)
 
+    def _get_targeted_move_ids(self):
+        return self.filtered(lambda c: c.target_model == 'picking').picking_ids.move_lines
+
+    def _get_targeted_records(self):
+        self.ensure_one()
+        return self.target_model == 'picking' and self.picking_ids or None
```

Comment:
we can remove this method i think, trust the onchange (because not trusting it result to a lot of ugly code)